posthog/test/test_feature_flags.py

```diff
@@ -1801,6 +1803,21 @@ def test_match_properties_math_operators(self):
 
         self.assertFalse(match_property(property_d, {"key": "44"}))
         self.assertFalse(match_property(property_d, {"key": 44}))
+        self.assertTrue(match_property(property_d, {"key": 42}))
+
+        property_e = self.property(key="key", value="30", operator="lt")
+        self.assertTrue(match_property(property_e, {"key": "29"}))
+
+        # depending on the type of override, we adjust type comparison
+        self.assertTrue(match_property(property_e, {"key": "100"}))
```

Comment:
shouldn't this be false??

Reply:
should we be converting these strings to numbers for the comparison

Reply:
or are we expecting users to expect this because they're passing in a string in both instances

Reply:
we can't always just convert, because we can't be sure it's a number (for example: 1.21 might be a version string, which can fail a numeric comparison, since 1.190 > 1.21)

So, we use the type of the override: if it's string, do a string comparison, if it's number, do a numeric comparison.